src/burn/engine/engine.cpp

```diff
@@ -864,52 +877,50 @@ static HRESULT RunApplication(
     // Unload BA.
     UserExperienceUnload(&pEngineState->userExperience, *pfReloadApp);
 
+    ::DeleteCriticalSection(&engineContext.csQueue);
+    ReleaseHandle(engineContext.hQueueSemaphore);
+    ReleaseQueue(engineContext.hQueue, FreeQueueItem, &engineContext);
```

Comment:
Just a thought as I got this far: A critical section and semaphore seem useful for many uses of `que`. Did you consider allowing those to be initialized as part of creating a queue? Then deleting the queue handle could clean all those up together. They could also be optional in the initialization. Just a thought based on usage I've seen thus far.

Reply:
I had the same idea but opted for simplicity. I think there are too many synchronization use cases to try to build it in and have a clean API.